### ISSUE TITLE
Updates for Products API 5?

### ISSUE THREAD

**Issue by mikeover** [Grantimus9/elixir-amazon-product-advertising-client#19]
I don't think this library will work anymore with Amazon Product Advertising API 5.0 and will be useless after the old version is discontinued on 3/9/2020. Any plans to update this library to work with 5.0?

**Comment by keslert** [comment]
Also interested in this as the deadline to switch to API 5.0 is March 9.

**Comment by Grantimus9** [comment]
I've spent a few hours on updating this, and have some work in progress. 

Amazon moving from PA4 to PA5 also means they are using an auth/signing system that's giving me some grief. On the plus side, most requests now amount to POSTing some JSON to their endpoints and their endpoints now return JSON instead of XML. 

I'm cautiously optimistic that this won't be as terrible of a migration for our package users as initially expected.

**Comment by keslert** [comment]
I made a little progress on an update as well. I didn't cover all the options, but maybe this will help? https://github.com/keslert/elixir-amazon-product-advertising-client

**Comment by Grantimus9** [comment]
Does your signature generation code work? I'm using the AWSAuth lib and getting pretty close (branch open on this repo with the basics). 

edit: if the signing code in your fork works, I'm all for adopting that, including the way you've setup the call_api call, and abandoning what I was doing.

**Comment by keslert** [comment]
Yes, it does. The biggest issue I had was that I was using the PHP sample code they provided and the order of parameters is reversed when using :crypto.hmac.

**Comment by dallagi** [comment]
Hi,

I'm not using this package, but I ran in the same problem while trying to authenticate some requests to the Product Advertising Api 5.

FYI I managed to successfully sign requests using the `aws_auth` package.

Just one note though: the version of that package from hex.pm won't work, as it has a bug that prevents it from working with the product advertising API.
I submitted a PR to fix it, and until it gets merged you can have a look at my fork [here](https://github.com/dallagi/aws_auth).

Hope this could be useful!

EDIT: I published a minimalistic library to build requests for the PAAPI5, you can find it here: https://github.com/dallagi/paapi5 . Feel free to take any code or inspiration from it, in case it may be useful.
